controllers/imagerepository_controller.go

```diff
@@ -500,13 +491,16 @@ func (r *ImageRepositoryReconciler) EnsureRemotePullSecret(ctx context.Context,
 			return err
 		}
 
+		serviceAccountName := buildPipelineServiceAccountName
+		if isPull {
```

Comment:
why are you using default SA for pull secret?

Reply:
Because there is no build pipeline SA in that namespaces and also no guarantee that any other SA exists.